### DISCUSSION TITLE
Significant more CPU usage after upgrade to 5.2.0

### DISCUSSION THREAD

**Discussion by joggeli34** [emqx/emqx#11590] [General]
Hi,

I updated our 3-node cluster running on kubernetes from 5.0.21 to 5.2.0-build.1. After the update, the sum of the used cpu-cores went up from 3 to 4, so it uses 33% more cpu than before.

![image](https://github.com/emqx/emqx/assets/34470368/521e8bff-f770-4960-9419-9f3ef7ee9aed)

We use http authz and authn, but otherwise it is more or less a standard installation with the helm chart. It serves around 3000 connections and 4000 messages per seconds. The log doesn't seem to be suspicious, as there are only very few entries. 

So my question is, if this is somehow expected with this update or is there something in the configuration that could be adjusted?

**Comment by qzhuyan** [comment]
could you also check the other metrics? such as packet rate? 
do you have wildcard subscriptions?

**Reply by joggeli34** [reply]
We have wildcard subscriptions as well as shared subscriptions. Package-rate is always the same, around 4000 messages per seconds. I just checked, it hasn't changed with the update.

Due to an other issue (https://github.com/emqx/emqx/issues/11593) we had to downgrade the cluster again to 5.0.21, and the cpu stays high as with the 5.2.0. So it probably has nothing to do with the version but with something else.

It could be due to how the clients are distributed to the nodes, but this should not lead to this much more cpu usage

**Reply by qzhuyan** [reply]
Ok. You are welcome to open new discussion if you have any clue. 

I close this one since the issue unrelates to version 5.2.0

**Comment by qzhuyan** [comment]
If I recall correctly, in 5.0.21, the rate limiter was ON by default, that is why I ask you to check the packet rate in the network.
